std/math.d

```diff
@@ -240,8 +240,350 @@ enum RealFormat
     ieeeQuadruple,
 }
 
-// Constants used for extracting the components of the representation.
-// They supplement the built-in floating point properties.
+/* Low-level access to the individual parts of a floating-point value.
+
+   TODO: Make this publically available through std.bitmanip once the API
+   is stable and tested.
+*/
+union RealRep(N)
+    if (isFloatingPoint!N &&  is(N == Unqual!N))
+{
+pure: @safe: nothrow: @nogc:
+pragma(inline, true):
+
+    static if (N.mant_dig == 11 && N.max_exp == 0x6)
+    {
+        enum format = RealFormat.ieeeHalf,
+             excessDig = 0,
+             storeWhole = false,
+             expDig = 5;
+        alias Mant = ushort,
+              Exp = byte;
```

Comment:
I am happy to document this, once we have agreed on the design. I just don't want to waste time writing docs that will be made irrelevant by a big rework.

As for longer names, in general I'd like to keep them reasonably short, but there is still some room to make them longer than they are now. Again, I'd like to leave that step until after the design is settled, though.

(Let me know if there's anything where you really just can't figure out what it's for, though, and I'll do something about it now. If you're able to offer a reasonable suggestion for a rename, that means you figured it out already.  :-)  )